src/activity_handlers.cpp

```diff
@@ -2187,33 +2184,66 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
             }
         }
 
+        // TODO: Allow setting this in the actor
+        // TODO: Don't use charges_to_use: welder has 50 charges per use, soldering iron has 1
+        if( !used_tool->ammo_sufficient() ) {
+            p->add_msg_if_player( _( "Your %s ran out of charges" ), used_tool->tname().c_str() );
```

Comment:
```suggestion
            p->add_msg_if_player( _( "Your %s ran out of charges" ), used_tool->tname() );
```
(I realize you didn't change this, only moved it, but boyscouting anyway)

Reply:
There are multiple usages of c_str() here with string_format as well. Might as well correct them as well. Will do them in a later commit.